components/roster/heroes/Heroes.tsx

```diff
@@ -0,0 +1,284 @@
+"use client";
+
+import { FormEvent, useReducer } from "react";
+import { Hero } from "@/lib/roster";
+import { Card, CardContent } from "@/components/ui/card";
+import { AscensionLevel } from "@/lib/characters";
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuLabel,
+  DropdownMenuSeparator,
+  DropdownMenuSub,
+  DropdownMenuSubContent,
+  DropdownMenuSubTrigger,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu";
+import { Checkbox } from "@/components/ui/checkbox";
+import { Label } from "@/components/ui/label";
+import HeroPortrait from "@/components/roster/heroes/HeroPortrait";
+import { Check, Star, Unlock, Lock } from "lucide-react";
+import Image from "next/image";
+import { rosterImages } from "@/lib/images";
+import { Button } from "@/components/ui/button";
+import { toast } from "sonner";
+import { cn } from "@/lib/utils";
+
+type SubSelectProps = {
+  hero: Hero;
+  heroProperty: "ascension" | "equipment";
+  options: AscensionLevel[] | string[];
+  selected: AscensionLevel | string | null | undefined;
```

Comment:
does this need all the types here?

maybe just `selected?: AscensionLevel | string` ?

why is the `string` needed for this and the above property?

Reply:
So the `string[]` is needed for the select menu part.. I couldn't get it to work without it. See `exclusiveEquipmentOptions` and `CustomSelectSubmenu`

Reply:
the `string` for AscensionLevel is needed because of the way it's stored in the db iirc. It's a number, and we can convert it to a string, and then `options` is an enum and the values are a string when you compare the two.